R/eml.R

```diff
@@ -29,6 +29,61 @@ pid_to_eml_other_entity <- function(mn, pids) {
   sysmeta_to_eml_other_entity(sysmeta)
 }
 
+#' Create an EML dataTable object for a given PID. 
+#' This function generates an attributeList and physical and constructs a dataTable.
+#'
+#' @param mn (MNode) Member Node where the PID is associated with an object.
+#' @param pid (character) The PID of the object to create the dataTable for.
+#' @param attributes (data.frame) Data frame of attributes.
+#' @param factors (data.frame) Data frame of enumerated attribute values (factors).
+#' @param name (character) Optional field to specify entityName, otherwise will be extracted from system metadata.
+#' @param description (character) Optional field to specify entityDescription, otherwise will match name.
+#' 
+#' @return (dataTable) The dataTable object
+#' @export
+#'
+#' @examples
+#' \dontrun{
+#' # Generate a dataTable for a given pid
+#' attributes <- create_dummy_attributes_dataframe(10)
+#' name <- "1234.csv"
+#' description <- "A description of this entity."
+#' dataTable <- pid_to_eml_datatable(mn, pid, attributes, name=name, description=description)
+#' }
+pid_to_eml_datatable <- function(mn, pid, attributes, factors=NULL, name=NULL, description=NULL) {
+  stopifnot(is(mn, "MNode"))
+  stopifnot(is.character(pid),
+            nchar(pid) > 0)
+  stopifnot(is.data.frame(attributes))
+  
+  if (is.null(factors)) {
+    attributes <- set_attributes(attributes)
+  } else {
+    stopifnot(is.data.frame(factors))
+    attributes <- set_attributes(attributes, factors)
+  }
+  
+  if (is.null(name)) {
+    name <- getSystemMetadata(mn, pid)@fileName
+  }
+  
+  if (is.null(description)) {
+    description <- name
+  }
+  
+  stopifnot(eml_validate_attributes(attributes))
```

Comment:
Cool you're using this. This is likely to save some headache. However, I see that the use of this isn't described in the documentation in the Details section. What do you think about documenting it there? And do you think a user would ever want to not run this? If so, an argument to skip it could be useful.